src/Symfony/Component/Notifier/Bridge/SmsBiuras/Tests/SmsBiurasTransportTest.php

```diff
@@ -44,4 +46,82 @@ public function unsupportedMessagesProvider(): iterable
         yield [new ChatMessage('Hello!')];
         yield [$this->createMock(MessageInterface::class)];
     }
+
+    /**
+     * @dataProvider testModeDataProvider()
+     */
+    public function testTestMode(array $provided, array $expected)
+    {
+        $message = new SmsMessage($provided['phone'], $provided['message']);
+
+        $response = $this->createMock(ResponseInterface::class);
+        $response->expects($this->atLeast(1))
+            ->method('getStatusCode')
+            ->willReturn(200);
+        $response->expects($this->atLeast(1))
+            ->method('getContent')
+            ->willReturn('OK: 519545');
+
+        $client = new MockHttpClient(function (string $method, string $url, array $options = []) use ($response, $expected): ResponseInterface {
+            $this->assertSame('GET', $method);
+            $this->assertSame(sprintf(
+                'https://savitarna.smsbiuras.lt/api?uid=uid&apikey=api_key&message=%s&from=from&test=%s&to=%s',
+                rawurlencode($expected['message']),
+                $expected['transport']['test_mode'],
+                rawurlencode($expected['phone']),
+            ), $url);
+            $this->assertSame($expected['transport']['test_mode'], $options['query']['test']);
+
+            $this->assertSame(200, $response->getStatusCode());
+            $this->assertSame('OK: 519545', $response->getContent());
+
+            return $response;
+        });
+
+        $transport = new SmsBiurasTransport('uid', 'api_key', 'from', $provided['transport']['test_mode'], $client);
+
+        $sentMessage = $transport->send($message);
+
+        $this->assertSame('519545', $sentMessage->getMessageId());
+    }
+
+    private function testModeDataProvider()
```

Comment:
Data providers should be public methods. And please use a static method as PHPUnit 10 will deprecate non-static data providers.